src/config/watcher.rs

```diff
@@ -19,11 +22,49 @@ const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1
 
 const RETRY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
 
+/// Refer to [`crate::cli::WatchConfigMethod`] for details.
+pub enum WatcherConfig {
```

Comment:
I wonder if we can refactor `create_watcher` to use the `WatchConfigMethod` enum directly. AFAIK, due to clap parsing, the `WatchConfigMethod` needs to have plain enum variants.

Reply:
Since i want that enum to have values about the interval and i don't want to pass the interval in the function parameters. 

One thing we can do is we can eliminate is watch config method enum. we can make it as string and add the possible values for `clap` arguments. using that we can create the `WatcherConfig` enum